### PR TITLE
Anonymous tracking

### PR DESCRIPTION
**Pull request by spenes**

### CODE REVIEW

File: http4s/src/test/scala/com.snowplowanalytics.snowplow.collectors.scalastream/CollectorServiceSpec.scala [hunk 1]
```diff
@@ -75,18 +78,170 @@ class CollectorServiceSpec extends Specification {
...
+          .unsafeRunSync()
+
+        r.status mustEqual Status.Ok
+        good.storedRawEvents must have size 1
+        bad.storedRawEvents must have size 0
+        val e = emptyCollectorPayload
+        deserializer.deserialize(e, good.storedRawEvents.head)
+        e.ipAddress shouldEqual "192.0.2.2"
+      }
+      "use the ip address from remote address if 'X-Forwarded-For' header doesn't exist" in {
```
Comment (pondzix) [inline http4s/src/test/scala/com.snowplowanalytics.snowplow.collectors.scalastream/CollectorServiceSpec.scala:181]: Isn't this test a duplicate of the one above?

File: http4s/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/CollectorService.scala [hunk 2]
```diff
@@ -109,7 +107,8 @@ class CollectorService[F[_]: Sync](
       ).flatten
       responseHeaders = Headers(headerList)
       _ <- sinkEvent(event, partitionKey)
-    } yield buildHttpResponse(responseHeaders, pixelExpected)
+      resp = buildHttpResponse(responseHeaders, pixelExpected)
+    } yield resp
```
Comment (pondzix) [inline http4s/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/CollectorService.scala:111]: Any specific reason why not inline like before? :)

File: http4s/src/test/scala/com.snowplowanalytics.snowplow.collectors.scalastream/CollectorServiceSpec.scala [hunk 3]
```diff
@@ -446,6 +597,90 @@ class CollectorServiceSpec extends Specification {
...
+        )
+        val expected = List(
+          "User-Agent: testUserAgent",
+          "X-Forwarded-For: 127.0.0.1",
+          "X-Real-Ip: 127.0.0.1",
+          "Cookie: cookie=value"
+        )
+        service.headers(request, None) shouldEqual expected
+      }
+      "filter out the headers if SP-Anonymous is not present" in {
```
[the remark this answers is not in the document]
Comment (spenes) [reply http4s/src/test/scala/com.snowplowanalytics.snowplow.collectors.scalastream/CollectorServiceSpec.scala:618]: Yep, good catch!

File: http4s/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/CollectorService.scala (hunk 2 above)
Comment (spenes) [reply http4s/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/CollectorService.scala:111]: No specific reason, I just like this more :D

**Review by pondzix — APPROVED** [review]

**Pull request merged**